### PR TITLE
Homework oop

### PR DESCRIPTION
**Pull request by gujaKhela**

### CODE REVIEW

File: src/Admin.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+public class Admin implements User {
+    @Override
+    public void printRole() {
+        System.out.println("I am an admin");
+    }
+
+    public void checkTicket() {
+        System.out.println("Admin: Checking ticket.");
+    }
```
Comment (Alexmansar) [inline src/Admin.java]: hane check logic expect some boolean value)

File: src/Client.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+public class Client implements User {
+    @Override
+    public void printRole() {
+        System.out.println("I am a client");
+    }
+
+    public void getTicket() {
+        System.out.println("Client: Getting ticket.");
+    }
```
Comment (Alexmansar) [inline src/Client.java:9]: and here expect just Ticket)

File: src/Identifiable.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+                    Object value = field.get(this);
+                    if (value == null) {
+                        System.out.println("Variable [" + field.getName() + "] is null in [" + this.getClass().getSimpleName() + "]!");
+                    }
+                } catch (IllegalAccessException e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+    }
```
Comment (Alexmansar) [inline src/Identifiable.java]: could be separate class

File: src/NullableWarning.java [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.FIELD)
+public @interface NullableWarning {
+}
```
Comment (Alexmansar) [inline src/NullableWarning.java:9]: 👍

File: src/Ticket.java [hunk 5]
```diff
@@ -1,31 +1,30 @@
...
-
     private double price;
     private final long creationTime;
 
-
-
-    public  Ticket() {
+    public Ticket() {
         this.creationTime = Instant.now().getEpochSecond();
+        checkNullableWarnings(); // Check nullable warnings in the constructor
```
Comment (Alexmansar) [inline src/Ticket.java]: this validation should be begore all this...
in your case - 18 line
And use it in every constructor

File: src/Identifiable.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
+import java.lang.reflect.Field;
+
+public abstract class Identifiable {
+    private String id;
```
Comment (Alexmansar) [inline src/Identifiable.java]: make it protected

File: src/Ticket.java [hunk 7]
```diff
@@ -125,16 +111,31 @@ private void setMaxBackpackWeight(double maxBackpackWeight) {
...
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(getId()); // Generate hash code based on ID
+    }
+
     @Override
     public String toString() {
-        return "Ticket [ID=" + id +
+        return "Ticket [ID=" + getId() +
```
Comment (Alexmansar) [inline src/Ticket.java]: and here will be just id

File: src/Ticket.java [hunk 8]
```diff
@@ -145,5 +146,24 @@ public String toString() {
...
+                    Object value = field.get(this);
+                    if (value == null) {
+                        System.out.println("Variable [" + field.getName() + "] is null in [" + this.getClass().getSimpleName() + "]!");
+                    }
+                } catch (IllegalAccessException e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+    }
```
Comment (Alexmansar) [inline src/Ticket.java]: remove it please. Just create 1 class)

**Review by Alexmansar — CHANGES_REQUESTED** [review]
@gujaKhela Hi, please, check small comments))

**Comment by gujaKhela** [comment]
resolved the feedback and pushed new commits

**Review by Alexmansar — APPROVED** [review]
@gujaKhela Hi, could be merged

**Pull request merged**